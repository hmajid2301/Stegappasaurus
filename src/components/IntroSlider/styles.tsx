import { StyleSheet } from "react-native";

import { fonts } from "~/modules";

const styles = StyleSheet.create({
  contentContainer: {
    flex: 1
  },

  container: {
    alignItems: "center",
    flex: 1,
    justifyContent: "space-around"
  },

  image: {
    height: 400,
    width: 400
  },

  title: {
    backgroundColor: "transparent",
    color: "white",
    fontFamily: fonts.header,
    fontSize: 35,
    textAlign: "center"
  },

  text: {
    backgroundColor: "transparent",
    color: "#F5F0F0",
    fontFamily: fonts.body,
    paddingBottom: 50,
    paddingHorizontal: 30,
    textAlign: "center"
  }
});

export default styles;
