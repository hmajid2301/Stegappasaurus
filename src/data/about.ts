import meta from '~/../app.json';
import {AboutItem} from '~/components/AboutList/AboutList';

const about = (color: any): AboutItem[] => [
  {
    icon: {
      color,
      name: 'versions',
      type: 'octicon',
    },
    title: `Version ${meta.version}`,
    url: 'https://github.com/hmajid2301/Stegappasaurus',
  },
  {
    icon: {
      color,
      name: 'code-fork',
      type: 'font-awesome',
    },
    title: 'Fork this Project on GitHub',
    url: 'https://github.com/hmajid2301/Stegappasaurus',
  },
  {
    icon: {
      color,
      name: 'gitlab',
      type: 'font-awesome',
    },
    title: 'Personal GitLab',
    url: 'https://gitlab.com/hmajid2301',
  },
  {
    icon: {
      color,
      name: 'web',
      type: 'material-community',
    },
    title: 'Personal Website',
    url: 'https://haseebmajid.dev',
  },
  {
    icon: {
      color,
      name: 'rate-review',
      type: 'material',
    },
    title: 'Rate the app',
  },
  {
    icon: {
      color,
      name: 'paypal',
      type: 'font-awesome',
    },
    title: 'Buy me a coffee',
    url: 'https://www.paypal.me/hmajid2301',
  },
];

export default about;
