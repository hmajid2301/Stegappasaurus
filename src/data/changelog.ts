const changelog = `
# Changelog

## [1.0.2] - 2020-02-23
Updated play store listing with new screenshots and featured graphic.
Also fixed small bugs when trying to share app and the action not being
logged correctly.

## [1.0.1] - 2020-02-01
Better error monitoring so that issues effecting user's of the app can be better tracked
and therefore resolved. It seems issues with encoding/decoding aren't currently being tracked correctly.
Can see issues in the app analytics but cannot see what is causing the errors.

## [1.0.0] - 2020-01-31
Initial Release.`;

export default changelog;
