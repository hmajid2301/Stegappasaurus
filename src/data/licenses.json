{
    "@react-native-community/async-storage": {
        "licenses": "MIT",
        "repository": "https://github.com/react-native-community/react-native-async-storage",
        "licenseUrl": "https://github.com/react-native-community/react-native-async-storage/raw/master/LICENSE",
        "parents": "stegappasaurus"
    },
    "@react-native-community/cameraroll": {
        "licenses": "MIT",
        "repository": "https://github.com/react-native-community/react-native-cameraroll",
        "licenseUrl": "https://github.com/react-native-community/react-native-cameraroll/raw/master/LICENCE",
        "parents": "stegappasaurus"
    },
    "bugsnag-react-native": {
        "licenses": "MIT",
        "repository": "https://github.com/bugsnag/bugsnag-react-native",
        "licenseUrl": "https://github.com/bugsnag/bugsnag-react-native/raw/master/LICENSE.txt",
        "parents": "stegappasaurus"
    },
    "lzutf8": {
        "licenses": "MIT",
        "repository": "https://github.com/rotemdan/lzutf8.js",
        "licenseUrl": "https://github.com/rotemdan/lzutf8.js/raw/master/LICENSE",
        "parents": "stegappasaurus"
    },
    "react-native-app-intro-slider": {
        "licenses": "MIT",
        "repository": "https://github.com/jacse/react-native-app-intro-slider",
        "licenseUrl": "https://github.com/jacse/react-native-app-intro-slider/raw/master/LICENSE",
        "parents": "stegappasaurus"
    },
    "react-native-circular-progress": {
        "licenses": "MIT",
        "repository": "https://github.com/bgryszko/react-native-circular-progress",
        "licenseUrl": "https://github.com/bgryszko/react-native-circular-progress/raw/master/LICENSE",
        "parents": "stegappasaurus"
    },
    "react-native-collapsible": {
        "licenses": "MIT",
        "repository": "https://oblador@github.com/oblador/react-native-collapsible",
        "licenseUrl": "https://oblador@github.com/oblador/react-native-collapsible/raw/master/LICENSE",
        "parents": "stegappasaurus"
    },
    "react-native-config": {
        "licenses": "MIT",
        "repository": "https://github.com/luggit/react-native-config",
        "licenseUrl": "https://github.com/luggit/react-native-config/raw/master/LICENSE",
        "parents": "stegappasaurus"
    },
    "react-native-elements": {
        "licenses": "MIT",
        "repository": "https://github.com/react-native-training/react-native-elements",
        "licenseUrl": "https://github.com/react-native-training/react-native-elements/raw/master/LICENSE",
        "parents": "stegappasaurus"
    },
    "react-native-gesture-handler": {
        "licenses": "MIT",
        "repository": "https://github.com/kmagiera/react-native-gesture-handler",
        "licenseUrl": "https://github.com/kmagiera/react-native-gesture-handler/raw/master/LICENSE",
        "parents": "stegappasaurus"
    },
    "react-native-image-picker": {
        "licenses": "MIT",
        "repository": "https://github.com/react-native-community/react-native-image-picker",
        "licenseUrl": "https://github.com/react-native-community/react-native-image-picker/raw/master/LICENSE.md",
        "parents": "stegappasaurus"
    },
    "react-native-markdown-renderer": {
        "licenses": "MIT",
        "repository": "https://github.com/mientjan/react-native-markdown-renderer",
        "licenseUrl": "https://github.com/mientjan/react-native-markdown-renderer/raw/master/LICENSE",
        "parents": "stegappasaurus"
    },
    "react-native-permissions": {
        "licenses": "MIT",
        "repository": "https://github.com/react-native-community/react-native-permissions",
        "licenseUrl": "https://github.com/react-native-community/react-native-permissions/raw/master/LICENSE",
        "parents": "stegappasaurus"
    },
    "react-native-rate": {
        "licenses": "MIT",
        "repository": "https://github.com/KjellConnelly/react-native-rate",
        "licenseUrl": "https://github.com/KjellConnelly/react-native-rate",
        "parents": "stegappasaurus"
    },
    "react-native-reanimated": {
        "licenses": "MIT",
        "repository": "https://github.com/kmagiera/react-native-reanimated",
        "licenseUrl": "https://github.com/kmagiera/react-native-reanimated/raw/master/LICENSE",
        "parents": "stegappasaurus"
    },
    "react-native-screens": {
        "licenses": "MIT",
        "repository": "https://github.com/kmagiera/react-native-screens",
        "licenseUrl": "https://github.com/kmagiera/react-native-screens/raw/master/LICENSE",
        "parents": "stegappasaurus"
    },
    "react-native-share": {
        "licenses": "MIT",
        "repository": "https://github.com/react-native-community/react-native-share",
        "licenseUrl": "https://github.com/react-native-community/react-native-share/raw/master/LICENSE",
        "parents": "stegappasaurus"
    },
    "react-native-snackbar": {
        "licenses": "MIT",
        "repository": "https://github.com/cooperka/react-native-snackbar",
        "licenseUrl": "https://github.com/cooperka/react-native-snackbar/raw/master/LICENSE",
        "parents": "stegappasaurus"
    },
    "react-native-splash-screen": {
        "licenses": "MIT",
        "repository": "https://github.com/crazycodeboy/react-native-splash-screen",
        "licenseUrl": "https://github.com/crazycodeboy/react-native-splash-screen/raw/master/LICENSE",
        "parents": "stegappasaurus"
    },
    "react-native-svg": {
        "licenses": "MIT",
        "repository": "https://github.com/react-native-community/react-native-svg",
        "licenseUrl": "https://github.com/react-native-community/react-native-svg/raw/master/LICENSE",
        "parents": "stegappasaurus"
    },
    "react-native-vector-icons": {
        "licenses": "MIT",
        "repository": "https://github.com/oblador/react-native-vector-icons",
        "licenseUrl": "https://github.com/oblador/react-native-vector-icons/raw/master/LICENSE",
        "parents": "stegappasaurus"
    },
    "react-native": {
        "licenses": "MIT",
        "repository": "https://github.com/facebook/react-native",
        "licenseUrl": "https://github.com/facebook/react-native/raw/master/LICENSE",
        "parents": "stegappasaurus"
    },
    "react-navigation-stack": {
        "licenses": "MIT",
        "repository": "https://github.com/react-navigation/react-navigation-stack",
        "licenseUrl": "https://github.com/react-navigation/react-navigation-stack/raw/master/LICENSE.md",
        "parents": "stegappasaurus"
    },
    "react-navigation-tabs": {
        "licenses": "MIT",
        "repository": "https://github.com/react-navigation/tabs",
        "licenseUrl": "https://github.com/react-navigation/tabs/raw/master/LICENSE.md",
        "parents": "stegappasaurus"
    },
    "react-navigation": {
        "licenses": "MIT",
        "repository": "https://github.com/react-navigation/react-navigation",
        "licenseUrl": "https://github.com/react-navigation/react-navigation/raw/master/LICENSE",
        "parents": "stegappasaurus"
    },
    "react": {
        "licenses": "MIT",
        "repository": "https://github.com/facebook/react",
        "licenseUrl": "https://github.com/facebook/react/raw/master/LICENSE",
        "parents": "stegappasaurus"
    },
    "rn-fetch-blob": {
        "licenses": "MIT",
        "repository": "https://github.com/joltup/rn-fetch-blob",
        "licenseUrl": "https://github.com/joltup/rn-fetch-blob/raw/master/LICENSE",
        "parents": "stegappasaurus"
    },
    "stream": {
        "licenses": "MIT",
        "repository": "https://github.com/juliangruber/stream",
        "licenseUrl": "https://github.com/juliangruber/stream",
        "parents": "stegappasaurus"
    },
    "utf8-to-bytes": {
        "licenses": "Apache-2.0",
        "repository": "git://gitlab.com/hmajid2301/utf8-to-bytes",
        "licenseUrl": "https://gitlab.com/hmajid2301/utf8-to-bytes",
        "parents": "stegappasaurus"
    },
    "varint": {
        "licenses": "MIT",
        "repository": "https://github.com/chrisdickinson/varint",
        "licenseUrl": "https://github.com/chrisdickinson/varint",
        "parents": "stegappasaurus"
    }
}
