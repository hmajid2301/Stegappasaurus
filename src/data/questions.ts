import {IFAQ} from '~/components/FAQList';

const questions: IFAQ[] = [
  {
    content: 'Yes the app is 100% free and fully open source.',
    title: 'Is Stegappasaurus mobile application free?',
  },
  {
    content: 'At the moment just Android.',
    title: 'What OS is the application available on?',
  },
  {
    content:
      "Simple answer no, all the data is stored on the user's (your device).",
    title: 'Do you store any data?',
  },
  {
    content: `Steganography is the art/science of hiding data in plain sight. In the context of this \
app, it allows you to hide messages (text) within images. You do this by encoding the image with \
the text data. You can then share the image with other people and they can decode the image \
(retrieve the original message) using this app.`,
    title: 'What is steganography?',
  },
  {
    content: `At the moment just LSB (least significant bit).`,
    title: 'What algorithms do you use?',
  },
  {
    content: `Not quite, encryption, involves obscuring information whereas steganography is concerned with hiding it. \
With encryption if you have the encrypted message you cannot retrieve the original message. \n\n \
Whereas with steganography if you know where to look you can retrieve the original message, it is \
not obscured in any way. However, with encryption, it is obvious you are trying to hide something \
but with steganography you can hide your message inside of an "innocent" looking image, with a third \
party without anyone noticing.\n\n \
You can combine the two "techniques", encrypt your data before encoding it into an image, then simply \
decrypt the message after decoding it from the image for added security. This involves a shared secret \
"password" between you and the person you are sending the image to.`,
    title: 'So is it the same as encryption?',
  },
  {
    content: `This app uses different steganography algorithms to encode the RGB (Red, Green, Blue) pixel values \
with the text data in such a way that the original data can be retrieved at a later date.\n\n \
After the image has been encoded you can save the image, then you can share this image and you use this app \
to decode and get the originally encoded message.\n\n \
In general each algorithm will encode the size of the message, this is so that when decoding the image we \
know when to stop. Each character in the message usually uses 8 bits in the images, this is because we \
encode it using the UTF-8 representation of the letter, symbol or emoji.`,
    title: 'How does this app work?',
  },
];

export default questions;
