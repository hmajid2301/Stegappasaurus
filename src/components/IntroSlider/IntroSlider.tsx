import React from "react";
import { Image, Text, View } from "react-native";
import AppIntroSlider from "react-native-app-intro-slider";

import { ISlide } from "@types";
import styles from "./styles";

interface IProps {
  slides: ISlide[];
  onDone: () => void;
}

interface IRenderProps {
  dimensions: {
    height: number;
    width: number;
  };
  item: ISlide;
  index: number;
}

export default class App extends React.Component<IProps, {}> {
  public render() {
    return (
      <AppIntroSlider
        onDone={this.props.onDone}
        onSkip={this.props.onDone}
        renderItem={this.renderSlide}
        showSkipButton
        slides={this.props.slides}
      />
    );
  }

  public renderSlide = ({ dimensions, item }: IRenderProps) => (
    <View
      style={[
        styles.container,
        {
          backgroundColor: item.color,
          height: dimensions.height,
          width: dimensions.width
        }
      ]}
    >
      <Text style={styles.title}>{item.title}</Text>
      <Image source={item.image} style={styles.image} />
      <Text style={styles.text}>{item.text}</Text>
    </View>
  );
}
