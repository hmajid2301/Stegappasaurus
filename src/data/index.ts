import about from "./about";
import questions from "./questions";
import slides from "./slides";

export { about, questions, slides };
