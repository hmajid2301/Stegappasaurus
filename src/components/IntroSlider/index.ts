import IntroSlider, {ISlide} from './IntroSlider';

export type ISlide = ISlide;
export default IntroSlider;
