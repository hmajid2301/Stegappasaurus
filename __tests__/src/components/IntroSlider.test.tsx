import {shallow} from 'enzyme';
import React from 'react';

import IntroSlider from '~/components/IntroSlider';
import {slides} from '~/data';

describe('IntroSlider: Match snapshots', () => {
  test('1', () => {
    const component = shallow(
      <IntroSlider slides={slides} onDone={() => null} />,
    );
    expect(component).toMatchSnapshot();
  });

  test('2', () => {
    const component = shallow(<IntroSlider slides={[]} onDone={() => null} />);
    expect(component).toMatchSnapshot();
  });
});
