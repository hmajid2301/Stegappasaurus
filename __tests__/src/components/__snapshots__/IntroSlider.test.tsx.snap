// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IntroSlider: Match snapshots 1 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <StatusBar
    animated={false}
    hidden={true}
    showHideTransition="fade"
  />
  <AppIntroSlider
    activeDotStyle={
      Object {
        "backgroundColor": "rgba(255, 255, 255, .9)",
      }
    }
    buttonStyle={null}
    buttonTextStyle={null}
    doneLabel="Done"
    dotStyle={
      Object {
        "backgroundColor": "rgba(0, 0, 0, .2)",
      }
    }
    nextLabel="Next"
    onDone={[Function]}
    onSkip={[Function]}
    paginationStyle={null}
    prevLabel="Back"
    renderItem={[Function]}
    showDoneButton={true}
    showNextButton={true}
    showSkipButton={true}
    skipLabel="Skip"
    slides={
      Array [
        Object {
          "color": "#06C49C",
          "image": 1,
          "key": "encoding",
          "text": "Hide your data within images.",
          "title": "Hide data. ",
        },
        Object {
          "color": "#129FDD",
          "image": 1,
          "key": "share",
          "text": "Then share the encoded images with your friends.",
          "title": "Share your images",
        },
        Object {
          "color": "#643FEA",
          "image": 1,
          "key": "decoding",
          "text": "Finally use this app to retrieve your hidden data from the image.",
          "title": "Retrieve hidden data",
        },
      ]
    }
  />
</View>
`;

exports[`IntroSlider: Match snapshots 2 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <StatusBar
    animated={false}
    hidden={true}
    showHideTransition="fade"
  />
  <AppIntroSlider
    activeDotStyle={
      Object {
        "backgroundColor": "rgba(255, 255, 255, .9)",
      }
    }
    buttonStyle={null}
    buttonTextStyle={null}
    doneLabel="Done"
    dotStyle={
      Object {
        "backgroundColor": "rgba(0, 0, 0, .2)",
      }
    }
    nextLabel="Next"
    onDone={[Function]}
    onSkip={[Function]}
    paginationStyle={null}
    prevLabel="Back"
    renderItem={[Function]}
    showDoneButton={true}
    showNextButton={true}
    showSkipButton={true}
    skipLabel="Skip"
    slides={Array []}
  />
</View>
`;

exports[`IntroSlider: Match snapshots 3 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <StatusBar
    animated={false}
    hidden={true}
    showHideTransition="fade"
  />
  <AppIntroSlider
    activeDotStyle={
      Object {
        "backgroundColor": "rgba(255, 255, 255, .9)",
      }
    }
    buttonStyle={null}
    buttonTextStyle={null}
    doneLabel="Done"
    dotStyle={
      Object {
        "backgroundColor": "rgba(0, 0, 0, .2)",
      }
    }
    nextLabel="Next"
    onDone={[Function]}
    onSkip={[Function]}
    paginationStyle={null}
    prevLabel="Back"
    renderItem={[Function]}
    showDoneButton={true}
    showNextButton={true}
    showSkipButton={true}
    skipLabel="Skip"
    slides={Array []}
  />
</View>
`;
